lib/loadbalance/loadbalancer_test.go

```diff
@@ -0,0 +1,309 @@
+package loadbalance
+
+import (
+	"context"
+	"io"
+	"net"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/KiaFarhang/tcp-load-balancer/internal/assert"
+)
+
+/*
+By default Go runs tests for a package sequentially, so we're fine sharing ports like this
+across tests. If we wanted to speed them up and run in parallel we'd have to get fancier (e.g.) dynamically
+generate them, but they're running in < .1 seconds as it is.
+*/
+const (
+	loadBalancerPort int = 4444
+	upstreamAPort    int = 5555
+	upstreamBPort    int = 6666
+)
```

Comment:
Great suggestion. For the sake of time, and since this current method works, I'm inclined to leave it as is and focus on working on the rest of the project. But in a more robust implementation I'd definitely want to do it this way. What do you think?